### PR TITLE
Add support for ULN repositories on new Zypper based reposync

### PR DESCRIPTION
**Pull request by isbm**
## What does this PR change?

Implements ULN authentication (reusable in Zypper plugin and RHN plugin)

## Test coverage
- Unit tests were added

## Links
- Fixes: https://github.com/SUSE/salt-board/issues/334

### CODE REVIEW

**Comment by isbm** [comment]
gitarro no changelog needed !!!

**Comment by isbm** [comment]
Still need to add zypper plugin at least skeleton for this. We anyway cannot yet properly test this without real credentials.

File: backend/satellite_tools/ulnauth.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        """
+        Get ULN token.
+
+        :raises RhnSyncException: if configuration does not contain required sections.
+        :returns: ULN token
+        """
+        usr, pwd = self._get_credentials()
+        px_url, px_usr, px_pwd = get_proxy(url)
+        hostname, label = self._get_hostname(url)
+        self._uln_url = "https://{}/XMLRPC/GET-REQ{}".format(hostname, label)
```
Comment (mcalmer) [inline backend/satellite_tools/ulnauth.py]: `self._uln_url` never used. Is this wanted ?

File: backend/satellite_tools/repo_plugins/uln_src.py [hunk 2]
```diff
@@ -22,55 +22,38 @@
 # pylint: disable=E0012, C0413
 import sys
 sys.path.append('/usr/share/rhn')
-from up2date_client.rpcServer import RetryServer, ServerList
 
 from spacewalk.satellite_tools.repo_plugins.yum_src import ContentSource as yum_ContentSource
 from spacewalk.satellite_tools.syncLib import RhnSyncException
+from spacewalk.satellite_tools.ulnauth import ULNAuth
 
 ULNSRC_CONF = '/etc/rhn/spacewalk-repo-sync/uln.conf'
```
Comment (mcalmer) [inline backend/satellite_tools/repo_plugins/uln_src.py]: I wonder if we can import this from ULNAuth class?
Currently we have this defined twice.

Comment (meaksh) [reply backend/satellite_tools/repo_plugins/uln_src.py]: Updated. I'll merge the PR once the tests are green.

**Pull request merged**